src/coreclr/vm/assembly.cpp

```diff
@@ -490,38 +487,24 @@ Assembly *Assembly::CreateDynamic(AssemblyBinder* pBinder, NativeAssemblyNamePar
 
     // Start loading process
     {
-        // Create a concrete assembly
-        // (!Do not remove scoping brace: order is important here: the Assembly holder must destruct before the AllocMemTracker!)
-        NewHolder<Assembly> pAssem;
-
+        if (createdNewAssemblyLoaderAllocator)
         {
             GCX_PREEMP();
-            // Assembly::Create will call SuppressRelease on the NewHolder that holds the LoaderAllocator when it transfers ownership
-            pAssem = Assembly::Create(pPEAssembly, pDomainAssembly->GetDebuggerInfoBits(), pLoaderAllocator->IsCollectible(), pamTracker, pLoaderAllocator);
-
-            ReflectionModule* pModule = (ReflectionModule*) pAssem->GetModule();
 
-            if (createdNewAssemblyLoaderAllocator)
-            {
-                // Initializing the virtual call stub manager is delayed to remove the need for the LoaderAllocator destructor to properly handle
-                // uninitializing the VSD system. (There is a need to suspend the runtime, and that's tricky)
-                pLoaderAllocator->InitVirtualCallStubManager();
-            }
+            // Initializing the virtual call stub manager is delayed to remove the need for the LoaderAllocator destructor to properly handle
+            // uninitializing the VSD system. (There is a need to suspend the runtime, and that's tricky)
+            pLoaderAllocator->InitVirtualCallStubManager();
         }
 
+        Assembly* pAssem = pDomainAssembly->GetAssembly();
         pAssem->m_isDynamic = true;
```

Comment:
I would move this right after `new DomainAssembly(...)`. It looks out of place here.